WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource+Sections.swift

```diff
@@ -0,0 +1,234 @@
+import Foundation
+import Yosemite
+
+
+// MARK: - Sections
+/// Order Details: TableView Section data.
+///
+extension OrderDetailsDataSource {
+    /// Setup: Sections
+    ///
+    /// CustomerInformation Behavior:
+    /// When: Customer Note == nil          >>> Hide Customer Note
+    /// When: Shipping == nil               >>> Display: Shipping = "No address specified"
+    ///
+    func reloadSections() {
+        let summary = Section(row: .summary)
+
+        let shippingNotice: Section? = {
+            //Hide the shipping method warning if order contains only virtual products or if the order contains only one shipping method
+            if isMultiShippingLinesAvailable(for: order) == false {
+                return nil
+            }
+
+            return Section(title: nil, rightTitle: nil, footer: nil, rows: [.shippingNotice])
+        }()
+
+        let products: Section? = {
+            guard items.isEmpty == false else {
+                return nil
+            }
+
+            var rows: [Row] = Array(repeating: .orderItem, count: items.count)
+            if isProcessingPayment {
+                rows.append(.fulfillButton)
+            } else {
+                rows.append(.details)
+            }
+
+            return Section(title: Title.product, rightTitle: Title.quantity, rows: rows)
+        }()
+
+        let customerInformation: Section = {
+            var rows: [Row] = []
+
+            if customerNote.isEmpty == false {
+                rows.append(.customerNote)
+            }
+            if order.shippingAddress != nil {
+                rows.append(.shippingAddress)
+            }
+            if shippingLines.count > 0 {
+                rows.append(.shippingMethod)
+            }
+            rows.append(.billingDetail)
+
+            return Section(title: Title.information, rows: rows)
+        }()
+
+        let payment: Section = {
+            var rows: [Row] = [.payment, .customerPaid]
+
+            if FeatureFlag.refunds.enabled {
```

Comment:
I believe we started using the feature flag service from the `ServiceLocator` that enables mocking, like `ServiceLocator.featureFlagService.isFeatureFlagEnabled(. refunds)` with the feature flag configured in the default implementation `DefaultFeatureFlagService`

Reply:
Thank you!